tv_apps/smart-system/js/sync_manager.js

```diff
@@ -0,0 +1,621 @@
+ * IAC requests or Settings API changes and translates these changes into
+ * actions. It makes use of the SyncStateMachine to control that these actions
+ * are possible on the current situation of Sync and communicates with the
+ * Sync application via IAC API messages when needed.
+ *
+ * The IAC based API that is exposed to the Settings app (or any other
+ * future consumer that requires to manage Sync) expects IAC messages with
+ * a single property 'name' that can have any of the following values:
+ *    'enable': requests to enable Sync.
+ *    'disable': requests to disable Sync.
+ *    'sync': requests a sync on demand. Although we have scheduled
+ *            synchronizations, we allow on demand sync requests.
+ *    'getInfo': gets the current information about the state of Sync
+ *               including the state (enabled, disabled, etc.), the
+ *               last time a successful sync happened and the last error
+ *               if there's any.
+ *
+ * It also triggers one message through the same IAC API with name
+ * 'onsyncchange' when something (state, error, lastSync) changes on Sync.
+ *
+ * SyncManager also communicates with the Sync app via IAC. Every time a
+ * synchronization is required (either because the requestSync timer fires or
+ * because the user requested a sync on demand), we wake the Sync app up and
+ * give it the required information to perform the synchronization that
+ * includes the URL of the Syncto server, a valid FxA assertion and its
+ * associated encryption keys and the list of collections to be synchronized.
+ */
+
+(function() {
+  const FXA_EVENT = 'mozFxAccountsUnsolChromeEvent';
+
+  const SYNC_TASK = 'gaia::system::firefoxsync';
+  const SYNC_REQUEST_IAC_KEYWORD = 'gaia::sync::request';
+  const SYNC_MANAGEMENT_API_IAC_KEYWORD = 'gaia-sync-management';
+
+  const COLLECTIONS = ['history', 'passwords'];
+
+  // Keys of asyncStorage persisted data.
+  const SYNC_STATE = 'sync.state';
+  const SYNC_STATE_ERROR = 'sync.state.error';
+  const SYNC_LAST_TIME = 'sync.lastTime';
+  const SYNC_USER = 'sync.user';
+
+  var SyncManager = function() {};
+
+  SyncManager.STATES = [];
+
+  SyncManager.SERVICES = [];
+
+  SyncManager.SETTINGS = [
+    /** READ ONLY SETTINGS **/
+
+    // The user can choose the collections she wants to sync. These settings
+    // indicates the user choice per collection.
+    'sync.collections.history.enabled',
+    'sync.collections.passwords.enabled',
+
+    // Setting any of these two settings to true will make the synchronization
+    // of the collection readonly. That means that we will only be retrieving
+    // data from Firefox Sync and we won't be pushing any of the local
+    // modifications to the collections source.
+    'sync.collections.history.readonly',
+    'sync.collections.passwords.readonly',
+
+    'sync.server.url',
+    'sync.scheduler.interval',
+    'sync.scheduler.wifionly'
```

Comment:
You need to set defaults for all these settings in `build/config/common-settings.json`